assets/src/blocks/tests/index.js

```diff
@@ -35,6 +49,24 @@ export const blockEditRender = ( name, settings ) => {
 			attributes={ block.attributes }
 			setAttributes={ noop }
 			user={ {} }
-		/>,
+		/>
 	);
 };
+
+/**
+ * adds EE custom categories to wp.blocks categories
+ */
+const addEspressoBlockCategories = () => {
+	const categories = getCategories();
+	if ( ! some( categories, { slug: 'event-espresso' } ) ) {
+		categories.push(
+			{
+				"slug": 'event-espresso',
+				"title": __( 'Event Espresso', 'event_espresso' )
+			}
+		);
+		setCategories( categories );
+	}
+};
+
+
```

Comment:
Why is this file being kept?  Also, I'm not sure why its in the `/tests` folder unless it was a misunderstanding of something I suggested.  My suggestion in a different comment is that it would be removed into the existing `src/blocks/test` folder (not `tests`) and inside a file named `helpers.js`.  Originally I think you had this at the path `src/blocks/helpers.js`.  Regardless, I still don't see the necessity of leaving this file in place since its not code that is being used and I doubt it ever _will_ be used.

Reply:
> I'm not sure why its in the /tests folder unless it was a misunderstanding of something I suggested. 

from your comment here: https://github.com/eventespresso/event-espresso-core/pull/479#discussion_r210332859

> tn3rb 7 days ago 
> which folder are you referring to?
> 
> nerrad 7 days ago 
> The blocks/tests folder since this is a helper for block tests.

--- --- ---

> Originally I think you had this at the path src/blocks/helpers.js. 

no the original path was: `assets/src/blocks/helpers/index.js` so after being asked to move it to the `blocks/tests` folder it ended up as `assets/src/blocks/tests/index.js`

> Regardless, I still don't see the necessity of leaving this file in place since its not code that is being used and I doubt it ever will be used.

I think it would be far far better to actually register the block as part of the test because that **proves** without a doubt that it is working properly with WP core, whereas simply doing a snapshot of the edit function is meaningless really... what does it prove? It certainly doesn't show that the block works. It could output a bunch of garbage and as long as the autogenerated snapshot matches, the test would pass. 

I assumed that you switched out the test to just be a shallow snapshot of the edit function because we weren't able to inject a custom category into the registerBlock function during testing, but that we would switch back if/when we figure out how to do so. That last point may depend on changes to Gutenberg which could include the pull request I made to add a filter that would allow custom categories to get injected. I mentioned to you that I was making a PR for this issue, so it's a bit surprising to hear you say that you doubt this code would ever be used, ie: why would I bother with a PR if I didn't plan on trying to proceed with fully registering blocks during testing!?!

I also assumed from this comment you made in chat:

> I just left it (with some changes I made from the mock tests I was trying) in case you wanted to keep it for another reason.

that it would be ok to leave in place for the time being. You did say that you felt the file was not needed anymore but it didn't sound too imperative, especially after the above comment. But you're right it is not doing anything at the moment, so I'll remove it.

Reply:
> from your comment here: #479 (comment)

Ya that was a typo.  I meant `blocks/test`.  Regardless you did as suggested.  I still think it would be a good pattern to either have helpers specifically for tests to go in a pattern such as `test/helpers/index.js` or `test/helpers.js`.  The former might be better because then its a bit more distinctive that this is a separate folder for test helpers.

> I think it would be far far better to actually register the block as part of the test because that proves without a doubt that it is working properly with WP core, whereas simply doing a snapshot of the edit function is meaningless really... what does it prove?

I dig your pov, but I differ in opinion for a couple reasons:

- registering block api is already covered by GB tests.  There no need to duplicate that functionality in our tests.
- userland testing will cover whether a block is actually registered and usable.  Automating that is better done via e2e tests (which we haven't implemented yet but I'd like to start doing at some point).
- I agree a snapshot is a bare minimum.  However, it does have utility in ensuring there are no syntax errors, that any defaults are as expected, and catching any unintended regressions on those defaults. 
- I mentioned in slack that more test coverage can be added on the editor block (you can do things such as prop change checks, various rendering scenarios etc, different snapshots for different props...) but that I'd leave it up to you whether you did that or not.  I'm satisfied a snapshot is sufficient on the top level block assuming any components composing the block are sufficiently unit tested.

> You did say that you felt the file was not needed anymore but it didn't sound too imperative, especially after the above comment.

ya the comment I made in slack that you referenced was simply acknowledging that you may decide to leave the file in place for another reason, assuming that reason might involve continuing to add additional test coverage using it.  Since that reason wasn't apparent, I suggested in my review to remove it.

It's great you did your pull request for having categories wrapped in a filter.  I vaguely recall that had been discussed as an option at some point but discarded in favor of being registered via the block registration api.  We'll see what happens.